src/legacy/core_plugins/data/public/timefilter/timefilter_service.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { UiSettingsClientContract } from 'src/core/public';
+import { TimeHistory, Timefilter, TimeHistoryContract, TimefilterContract } from './index';
+
+/**
+ * Filter Service
+ * @internal
+ */
+
+export interface TimeFilterServiceDependencies {
+  uiSettings: UiSettingsClientContract;
+}
+
+export class TimefilterService {
+  public setup({ uiSettings }: TimeFilterServiceDependencies): TimefilterSetup {
+    const timefilterConfig = {
```

Comment:
hmm not sure if this is (was) working fine:
- you start kibana, TimefilterService gets started and reads the uiSettings
- you go to management -> kibana advanced settings and you change them
- they don't apply until you refresh your browser (to reload kibana)